components/script/dom/htmlimageelement.rs

```diff
@@ -976,6 +1195,20 @@ impl MicrotaskRunnable for ImageElementMicrotask {
                     elem.update_the_image_data_sync_steps();
                 }
             },
+            &ImageElementMicrotask::EnvironmentChangesTask { ref elem, ref generation } => {
+                // Step 2 of https://html.spec.whatwg.org/multipage/#img-environment-changes,
+                let element = elem.upcast::<Element>();
+                let has_src = element.has_attribute(&local_name!("srcset"));
+                let document = document_from_node(element);
+                let has_pending_request =  match elem.image_request.get() {
+                    ImageRequestPhase::Pending => true,
+                    _ => false
+                };
+                // TODO Handle image/x-mixed-replace
+                if document.is_active() && has_src && !has_pending_request {
```

Comment:
We should remove these checks now, since the exist inside react_to_environment_changes_sync_steps.